### PR TITLE
3.1 - Autoswitch elements to prefix path

### PR DESCRIPTION
**Pull request by HavokInspiration**
As Layouts and action view files already switch to a prefix path if available (`src/Template/{Prefix}/Layout/my_layout.ctp` for instance), elements should too.
This PR implements this behavior.
It should be noted that this also autoswitch Flash elements.

More details on #4320

### CODE REVIEW

File: tests/test_app/TestApp/Template/FooPrefix/BarPrefix/Element/prefix_element.ctp [hunk 1]
```diff
@@ -0,0 +1 @@
+<?= 'this is a nested prefixed test element'; ?>
```
Comment (markstory) [inline tests/test_app/TestApp/Template/FooPrefix/BarPrefix/Element/prefix_element.ctp]: Why are you echoing a string? Also these files need trailing newlines.

Comment (HavokInspiration) [inline tests/test_app/TestApp/Template/FooPrefix/BarPrefix/Element/prefix_element.ctp]: No particular reason.
I duplicated a flash element file from the TestPlugin provided with the tests suite and just changed the output ([link to file](https://github.com/cakephp/cakephp/blob/3.1/tests/test_app/Plugin/TestPlugin/src/Template/Element/Flash/plugin_element.ctp)).
I'll change it to make it more "real life example".

And ok for new lines, I'll add them.

**Comment by HavokInspiration** [comment]
@markstory I added a new line to the added files.
Do you want me to remove the string echoing ? If so, I'll do the already existing one along the way.

**Comment by jadb** [comment]
I would like to suggest adding this to 3.0.x, for same reasons noted [here](https://github.com/cakephp/cakephp/pull/6210#issuecomment-87937338).

**Comment by HavokInspiration** [comment]
@jadb I surely can rebase this to 3.0.x.

File: tests/test_app/TestApp/Template/FooPrefix/BarPrefix/Element/prefix_element.ctp (hunk 1 above)
Comment (ADmad) [inline tests/test_app/TestApp/Template/FooPrefix/BarPrefix/Element/prefix_element.ctp]: No point echoing static strings. Please just put it as text.

**Comment by ADmad** [comment]
:+1: to merging this to 3.0

**Comment by HavokInspiration** [comment]
Closing in favor of #6233

**Pull request closed without merging**